src/pydicom/encaps.py

```diff
@@ -866,6 +1156,62 @@ def encapsulate(
     return bytes(output)
 
 
+def encapsulate_buffer(
+    buffers: list[BufferedIOBase], has_bot: bool = True
+) -> EncapsulatedBuffer:
+    """Return an :class:`~pydicom.encaps.EncapsulatedBuffer` instance from `buffers`.
+
+    .. versionadded:: 3.0
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        from pydicom import Dataset, FileMetaDataset
+        from pydicom.encaps import encapsulate_buffer
+        from pydicom.uid import JPEG2000Lossless
+
+        # Open the compressed image frames as io.BufferedReader instances
+        frame1 = open("frame1.j2k", "rb")
+        frame2 = open("frame2.j2k", "rb")
+        frame3 = open("frame3.j2k", "rb")
```

Comment:
I would have used context managers in real code, but I agree that the indentation levels may look awkward here